### ISSUE TITLE
warning about runtime check from dynamic

### ISSUE THREAD

**Issue by jmesserly** [dart-lang/dev_compiler#368]
Is this supposed to be a warning?
I don't mind fixing it, just seems odd.
```
Warning:(465, 41) ts (dynamic) will need runtime check to cast to type List<TypeName>
```
code:
```dart
  TypeName _typeNameFromFunctionType(FunctionType type) {
    if (_typedefs.containsKey(type)) {
      var alias = _typedefs[type];
      var ts = null;
      var tpl = alias.typeParameters;
      if (tpl != null) {
        var ltp = tpl.typeParameters;
        ts = new List<TypeName>.from(
            ltp.map((t) => _mkNewTypeName(null, t.name, null)));
      }
      var name = alias.name;
      return _mkNewTypeName(type, name, ts); // WARNING HERE
    }

    // ...
  }
```

**Comment by jmesserly** [comment]
I understand `List<dynamic>` to `List<String>` ... I guess here too, we're worried it could be a `List<dynamic>`? Maybe we can link to some documentation about why this warning exists? Otherwise it feels like it's just complaining about me using `dynamic`, which then feels inconsistent (why complain here, and not elsewhere)

**Comment by vsmenon** [comment]
Yeah, exactly.  We're emitting a warning as we think there is a high likelihood of runtime failure.  This one is a false positive though.

Does the analyzer use links today to more detailed explanations?

**Comment by leafpetersen** [comment]
I'd guess this pattern is too common to warrant a warning - we should probably only issue this warning if both sides of the cast are non-ground.

**Comment by jmesserly** [comment]
merging with https://github.com/dart-lang/sdk/issues/25255
